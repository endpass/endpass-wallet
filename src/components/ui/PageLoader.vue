<template>
  <div v-if="isLoading" class="page-loader modal is-active">
    <div class="modal-content">
      <div class="page-loader-logo">
        <img src="/static/logo-dark.png" alt="Endpass">
      </div>
      <div class="spinner">
        <div class="spinner-dot spinner-dot-1">
        </div>
        <div class="spinner-dot spinner-dot-2">
        </div>
        <div class="spinner-dot spinner-dot-3">
        </div>
      </div>
      <p class="spinner-caption">Loading...</p>
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex';
// Loading indicator that covers the entire page
export default {
  name: 'page-loader',
  computed: {
    ...mapState({
      isLoading: state => state.isPageLoading,
    }),
  },
};
</script>

<style lang="scss">
.page-loader {
  color: $white;
  background-color: $purple;

  .page-loader-logo {
    margin: 0 auto;
    text-align: center;
    img {
      height: auto;
      width: auto;
      max-height: 6rem;
    }
  }
  .spinner-caption {
    text-align: center;
    margin: 0 auto;
  }
  .spinner {
    margin: 1rem auto 0;
    width: 70px;
    text-align: center;

    .spinner-dot {
      width: 18px;
      height: 18px;
      background-color: $white;

      border-radius: 100%;
      display: inline-block;
      animation: bouncedelay 1.4s infinite ease-in-out both;
    }

    .spinner-dot-1 {
      animation-delay: -0.32s;
    }
    .spinner-dot-2 {
      animation-delay: -0.16s;
    }
  }
  @keyframes bouncedelay {
    0%,
    80%,
    100% {
      transform: scale(0);
    }
    40% {
      transform: scale(1);
    }
  }
}
</style>
