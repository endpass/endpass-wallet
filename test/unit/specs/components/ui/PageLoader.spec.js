import { shallow, createLocalVue } from '@vue/test-utils';

import Vuex from 'vuex';
import PageLoader from '@/components/ui/PageLoader';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('PageLoader page', () => {
  let wrapper, state;

  const storeOptions = {
    state: {
      isPageLoading: true,
    },
  };
  const store = new Vuex.Store(storeOptions);
  beforeEach(() => {
    wrapper = shallow(PageLoader, {
      localVue,
      store,
    });
  });

  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('PageLoader');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
