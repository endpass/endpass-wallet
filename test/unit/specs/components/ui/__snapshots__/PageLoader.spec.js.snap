// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader page render should render initial state of the component 1`] = `
<div
  class="page-loader modal is-active"
  data-test="page-loader"
>
  <div
    class="modal-content"
  >
    <div
      class="page-loader-logo"
    >
      <img
        alt="Endpass"
        src="/static/logo-dark.png"
      />
    </div>
     
    <div
      class="spinner"
    >
      <div
        class="spinner-dot spinner-dot-1"
      />
       
      <div
        class="spinner-dot spinner-dot-2"
      />
       
      <div
        class="spinner-dot spinner-dot-3"
      />
    </div>
     
    <p
      class="spinner-caption"
    >
      Loading...
    </p>
  </div>
</div>
`;
